Input Files for LA100.Socioeconomics_India

ankurmalyan opened the pull request:
The input files have been prepared for population, GDP, states_subregions, etc. for Indian states. Please review if these files are fine. 

Additionally, We have started working on R code as well which we'll share with you soon for review.

ankurmalyan committed:
Input Files for LA100.Socioeconomics_India

The input files has been prepared for population, GDP, states_subregions etc. for India.

ankurmalyan committed:
Updated input files for Socioeconomics GCAMINDIA

All the Union Territories except of Delhi are clubed into one category. Additionally there are some changes based on the assumptions.

rynacui commented:
Thank you @ankurmalyan. Please create one PR for each chunk. We usually do not do multiple PRs but possible multiple commits (adding input files can be one commit) with one PR for each chunk. 

Also, please give me writing access to the repository.

rynacui reviewed (commented):
Looks great overall! Please remove all the comma at the end of most the lines.

rynacui commented on input/gcamdata/inst/extdata/gcam-india/PRIMA_pop_India.csv:
```diff
@@ -0,0 +1,78 @@
+# Title: 2005-2100 population projections by state,,,,,,,,,,,,,,,,,,,,,
+# Units: Persons,,,,,,,,,,,,,,,,,,,,,
+# Source: Interpolation from the IIASA estimates,,,,,,,,,,,,,,,,,,,,,
+state,state_name,2005,2010,2015,2020,2025,2030,2035,2040,2045,2050,2055,2060,2065,2070,2075,2080,2085,2090,2095,2100
```
need column type here

ankurmalyan replied:
Sure. We'll remove the commas from the terminals.

ankurmalyan replied:
@rynacui, I had a query regarding the row "#------" which is present in all the files. What exactly it corresponds to and what is its role? and Are the no. of '-' are corresponding to the no. of columns?

rynacui commented on input/gcamdata/inst/extdata/gcam-india/SE_PopH_Census_1961_2011.csv:
```diff
@@ -0,0 +1,43 @@
+# File: SE_PopH_Census_1961_2011.csv,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```
Why are there two historical population files?

ankurmalyan replied:
We have prepared another set of input files with some updates:
1. We have grouped all the union territories under one head.
2. We have changed some assumptions to estimate the data of the newly formed state about which we were discussing in the last call.
3. We have standardized our structure of naming the files so that it could be easy for us to understand the file from its title. The new naming format which we are following now is:
Module_Sector_Subsector_Data_Source_Years
eg: Here in SE_PopH_Census_1961_2011, SE is Socioeconomics and PopH is population historical from 1961 to 2011.

ankurmalyan commented:
> Thank you @ankurmalyan. Please create one PR for each chunk. We usually do not do multiple PRs but possible multiple commits (adding input files can be one commit) with one PR for each chunk.
> 
> Also, please give me writing access to the repository.

Thanks a lot, @rynacui for reviewing the files. We'll make sure that we have one PR  for each chunk. I'll provide you with access to writing the repository

ankurmalyan committed:
Merge branch 'master' into MalyanA-LA100.Socioeconomics_India

ankurmalyan committed:
Socioeconomics_chunk for India

It estimtaes the state wise GDP, GDp per capita and Projected Population.

ankurmalyan committed:
Changes in Input files for socioeconomics chunk India

There are a few changes in socioeconomics chunk for India

1. Instead of multiple GDP input files we have prepared one.
2. All the union territrories are grouped into one.

ankurmalyan committed:
Revised LA100.Socioeconomics_India

There are a few improvements in the code to maintain the uniformity of format with GCAM-USA outputs.

ankurmalyan committed:
Discarded old files

Deleted old files. New files have already been uploaded and used

ankurmalyan committed:
changes in constants file and state_subregion file

ankurmalyan committed:
Socioeconomics_INDIA_XML

The chunk for generating final socioeconomics xml.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+             "L100.Pop_thous_state_india"))
+  } else if(command == driver.MAKE) {
+
+    state <- state_name <- year <- value  <- Area <- population <- iso <-
```
missing `total`

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      mutate(value = approx_fun(year, value, rule = 2)) ->
+      SE_SGDP_MOSPI_11INR
+
+    #Estimating the shares of state GDP
```
I think these two pipe sections can be combined. Any reason for breaking into two parts?

ankurmalyan replied:
Yes, Ryna. These two sections can be combined. The only reason to write them separately is to have a clear differentiation between the interpolating section and share estimating section.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+    SE_SGDP_MOSPI_11INR %>%
+      ungroup %>%
+      group_by(year) %>%
+      mutate(share = value/sum(value)) %>%
```
Space around `/` please

ankurmalyan replied:
Sure will add space around the arithmetic operators.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+    #Estimating state wise GDP @1990USD prices
+    ##FIltering the data for India
+    L100.gdp_mil90usd_ctry_Yh %>%
+      filter(iso == "ind") %>% select(year, total = value) ->  L100.gdp_mil90usd_India
```
Return to another row after `%>%` please

ankurmalyan replied:
Sure will do it in separate rows.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      left_join_error_no_match(L100.gdp_mil90usd_India, by = "year") %>%
+      mutate(value = share *total) %>%
+      select(-total, -share, -state_name) %>%
+      ungroup %>%
```
Please `ungroup` in the same section where the data is grouped.

ankurmalyan replied:
Okay. Yeah will take care of this and make changes in the chunks

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      mutate(value = share *total) %>%
+      select(-total, -share, -state_name) %>%
+      ungroup %>%
+      add_title("GDP by state") %>%
```
The section of adding output attributes is preferred to be placed at the end.

ankurmalyan replied:
Sure. will take care of that.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+    SE_PopH_Census_1961_2011 %>%
+      gather_years %>%
+      PH_year_value_historical %>%
+      rename (population = value)->
```
No space around `(` please

ankurmalyan replied:
Sure. Will remove spaces from there.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      add_title("Per-capita GDP by state") %>%
+      add_units("thousand 1990 USD per capita") %>%
+      add_comments("State wise GDP per capita thousands@1990USD from 1971 to 2010") %>%
+      add_precursors("L100.GDP_mil90usd_state_india") %>%
```
Precursor can only be one of the input files read in for this chunk.

ankurmalyan replied:
I didn't get your point. Will discuss this during the call.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      ungroup %>%
+      add_title("GDP by state") %>%
+      add_units("million 1990 USD") %>%
+      add_comments("State wise GDP Millions@1990USD prices from 1971 to 2010") %>%
```
Please replace `@`

ankurmalyan replied:
Sure.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      select(-state_name, -population) ->
+      L100.Pop_ratio_state
+
+    #Starting from end of history, projecting state populations into future
```
Please add more details in the comment about the method used to project future population

ankurmalyan replied:
Sure. Will make it more detailed.

rynacui commented on input/gcamdata/R/zchunk_LA100.Socioeconomics_India.R:
```diff
@@ -0,0 +1,143 @@
+      group_by(state) %>%
+      mutate(population = approx_fun(year, population)) %>%
+      arrange(state, year) %>%
+      mutate(pop_ratio = population / first(population)) %>%
```
I would suggest `mutate(pop_rate = population / population[year == socioeconomics.FINAL_HIST_YEAR])`, and drop row 108.

ankurmalyan replied:
Yes, we can replace the command with your suggested one.

rynacui commented on input/gcamdata/R/zchunk_L201.socioeconomics_INDIA.R:
```diff
@@ -0,0 +1,132 @@
+             "L100.GDP_mil90usd_state_india",
+             "L102.pcgdp_thous90USD_GCAM3_ctry_Y"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L201.InterestRate_GCAMINDIA",
```
Please have consistent naming for all GCAM-India csv outputs. `_india` is used for the level1 chunk.

ankurmalyan replied:
Sure. Will follow -India for Level 2 and _India for level 1

rynacui reviewed (commented):
Looks good!

ankurmalyan committed:
Revision in the socioeconomics codes

As per the discussion, I have revised the codes and have added the explanation wherever required. I have also added the detailed explanation on the estimation of future population.

ankurmalyan committed:
INDIA to India in XML

ankurmalyan committed:
Input Files for Energy Balance Chunk

Currently the national IEA's balance for India is equally proportioned among states and is being used as the representative data for the setting up of the chunk. The representative data will be updated during the development of further chunks.

ankurmalyan committed:
Energy Balance chunk for India

Energy Balance chunk for India to be reviewed.

ankurmalyan committed:
Addition of conversion constant from MToe to EJ

Added a constant to convert Mtoe values to EJ.

ankurmalyan committed:
Input File for LA119.solar_INDIA

This files contains capacity factors for utility scale PV, rooftop PV, CSP and wind for Indian states. The comments have been added to explain the estimations and assumptions taken for this file.

ankurmalyan committed:
R Chunk for LA119.solar_INDIA

The chunk has been modified for India. Details and comments have been added to explain the steps which are being considered to generate output files.

ankurmalyan committed:
Addition of price convertor and wind base cost year constant

ankurmalyan committed:
Merge branch 'LA119.solar_India' into MalyanA-LA100.Socioeconomics_India

ankurmalyan committed:
Merge branch 'LA114.wind_INDIA' into MalyanA-LA100.Socioeconomics_India

ankurmalyan committed:
Changes in constants and Socioeconomcis documentation

Added a new constant to convert 2019INR to 1975USD
